src/AppBundle/EventListener/RedirectToPreferredLocaleListener.php

```diff
@@ -42,38 +42,33 @@ class RedirectToPreferredLocaleListener
      */
     public function __construct(UrlGeneratorInterface $urlGenerator, $locales, $defaultLocale = null)
     {
-        $this->locales = explode('|', trim($locales));
+        $this->urlGenerator = $urlGenerator;
 
+        $this->locales = explode('|', trim($locales));
         if (empty($this->locales)) {
-            throw new \UnexpectedValueException('Locales argument value must not be empty');
+            throw new \UnexpectedValueException('The list of supported locales must not be empty.');
         }
 
-        $this->defaultLocale = $defaultLocale;
+        $this->defaultLocale = $defaultLocale ? $this->locales[0];
```

Comment:
@voronkovich with `$this->defaultLocale = $defaultLocale ?: $this->locales[0];` work.